packages/lesswrong/components/ea-forum/voting-portal/EAVotingPortalSubmitPage.tsx

```diff
@@ -1,59 +1,147 @@
-import React from "react";
+import React, { useCallback, useState } from "react";
 import { Components, registerComponent } from "../../../lib/vulcan-lib";
 import { AnalyticsContext } from "../../../lib/analyticsEvents";
 import { votingPortalStyles } from "./styles";
 import { useCurrentUser } from "../../common/withUser";
 import { isAdmin } from "../../../lib/vulcan-users";
 import { useNavigate } from "../../../lib/reactRouterWrapper";
+import TextField from "@material-ui/core/TextField";
+import classNames from "classnames";
+import { useElectionVote } from "./hooks";
 
 const styles = (theme: ThemeType) => ({
   ...votingPortalStyles(theme),
+  headingMargin: {
+    marginBottom: 26,
+  },
+  fullWidth: {
+    width: "100%",
+  },
+  explanationRow: {
+    width: "100%",
+    marginBottom: 32,
+  },
+  questionTitle: {
+    fontSize: 16,
+    fontWeight: 600,
+    marginBottom: 12,
+  },
+  textField: {
+    width: "100%",
+    "& .MuiInputBase-root": {
+      padding: 16,
+    },
+    "& textarea": {
+      color: theme.palette.grey[1000],
+      zIndex: theme.zIndexes.singleColumnSection,
+      fontSize: 14,
+    },
+    "& .MuiNotchedOutline-focused": {
+      border: `2px solid ${theme.palette.givingPortal[1000]} !important`,
+    },
+    "& .MuiNotchedOutline-root": {
+      backgroundColor: theme.palette.background.contrastInDarkMode,
+    },
+  },
+  greyedOut: {
+    color: theme.palette.grey[600],
+  },
 });
 
-// TODO: implement
-const EAVotingPortalSubmitPage = ({classes}: {classes: ClassesType}) => {
+const EAVotingPortalSubmitPageLoader = ({ classes }: { classes: ClassesType }) => {
+  const { electionVote, updateVote } = useElectionVote("givingSeason23");
+
+  if (!electionVote?.vote) return null;
+
+  return (
+    <EAVotingPortalSubmitPage
+      electionVote={electionVote}
+      updateVote={updateVote}
+      classes={classes}
+    />
+  );
+};
+
+const EAVotingPortalSubmitPage = ({
+  electionVote,
+  updateVote,
+  classes,
+}: {
+  electionVote: ElectionVoteInfo;
+  updateVote: (newVote: NullablePartial<DbElectionVote>) => Promise<void>;
+  classes: ClassesType;
+}) => {
   const { VotingPortalFooter } = Components;
   const navigate = useNavigate();
+  const currentUser = useCurrentUser();
+
+  const [userExplanation, setUserExplanation] = useState<string>(electionVote.userExplanation ?? "");
+  const [userOtherComments, setUserOtherComments] = useState<string>(electionVote.userOtherComments ?? "");
+
+  const handleSubmit = useCallback(async () => {
+    await updateVote({ userExplanation, userOtherComments, submittedAt: new Date() });
```

Comment:
I think this will mean time zone issues?

Reply:
You're right, I've made a followup ticket for [this](https://app.asana.com/0/0/1206054023475652/f) as it's not blocking testing the rest of it